tests/drop_counters/test_drop_counters.py

```diff
@@ -247,17 +269,23 @@ def base_verification(discard_group, pkt, ptfadapter, duthost, combined_counter,
         pytest.fail("Incorrect 'discard_group' specified. Supported values: 'L2' or 'L3'")
 
 
-def test_equal_smac_dmac_drop(ptfadapter, duthost, setup, tx_dut_ports, pkt_fields):
+def test_equal_smac_dmac_drop(ptfadapter, fanouthost, duthost, setup, tx_dut_ports, pkt_fields):
     """
     @summary: Verify that packet with equal SMAC and DMAC is dropped and L2 drop cunter incremented
     """
     dut_iface, ptf_tx_port_id, dst_mac, src_mac = get_test_ports_info(ptfadapter, duthost, setup, tx_dut_ports)
+    src_mac = "00:00:00:00:00:11"
 
     log_pkt_params(dut_iface, dst_mac, dst_mac, pkt_fields["ip_dst"], pkt_fields["ip_src"])
+    
+    if "mellanox" == duthost.facts["asic_type"]:
+        # Prepare openflow rule
+        fanouthost.update_config(template_path=os.path.join(os.path.dirname(__file__),
+                                    "fanout/mellanox/mlnx_update_smac.j2"), match_mac=src_mac, set_mac=dst_mac)
```

Comment:
else what?

Reply:
Fixed. Now default behavior is that dmac assigns to smac (smac=smac).
If there is a need for specific vendor they can change behavior.
Here in Mellanox case - fanout configuration and smac value are updated, for other vendors the behavior is default.